lib/rihanna/job.ex

```diff
@@ -604,4 +614,8 @@ defmodule Rihanna.Job do
   defp producer_query({Postgrex, conn}, query, args) do
     Postgrex.query(conn, query, args)
   end
+
+  defp producer_query(_connection, query, args) do
```

Comment:
This is confusing. I'd prefer to be explicit about it. If someone has passed in an incorrect option we ought to raise, not silently ignore.